doc/source/coding_style.rst

```diff
@@ -135,3 +135,173 @@ update any projects dependent on the API while still being treated as
+each ``doctest`` session.
+
+.. code:: python
+
+    import pytest
+
+    from pyaedt import Desktop
+
+
+    @pytest.fixture(autouse=True, scope="session")
+    def start_aedt():
+        desktop = Desktop("2021.1", NG=True)
+        desktop.disable_autosave()
+
+        # Wait to run doctest on docstrings
+        yield desktop
+        desktop.force_close_desktop()
+
+Fixtures can also be defined in a separate Python file from 
+``conftest.py``. This may help keep the fixtures more organized. Fixtures 
+from other files need to be imported in the main ``conftest.py`` file. The 
+following example shows how to import fixtures defined in an 
+``icepak_fixtures.py`` file under the ``doctest_fixtures`` folder.
+
+.. code:: python
+
+    import pytest
+
+    from pyaedt import Desktop
+    from pyaedt.doctest_fixtures import *
+
+    # Import fixtures from other files
+    pytest_plugins = [
+        "pyaedt.doctest_fixtures.icepak_fixtures",
+    ]
+
+
+    @pytest.fixture(autouse=True, scope="session")
+    def start_aedt():
+        desktop = Desktop("2021.1", NG=True)
+        desktop.disable_autosave()
+
+        # Wait to run doctest on docstrings
+        yield desktop
+        desktop.force_close_desktop()
+
+The ``doctest_namespace`` fixture built in to ``doctest`` allows injecting
+items from a fixture into the context of the ``doctest`` run. To use this
+feature, the fixture needs to accept the ``doctest_namespace`` dictionary
+as an argument. Then, objects can be added to the ``doctest_namespace``
+dictionary and used directly in a docstring example.
+
+The code below shows how the ``Icepak`` object can be stored in the
+``doctest_namespace`` dictionary by adding the key ``icepak`` with the 
+``Icepak`` object as the value. 
+
+.. code:: python
+
+    import pytest
+    from pyaedt import Icepak
+
+
+    @pytest.fixture(autouse=True, scope="module")
+    def create_icepak(doctest_namespace):
+        doctest_namespace["icepak"] = Icepak(projectname="Project1", designname="IcepakDesign1")
+
+Then, the ``Icepak`` object can be used directly inside a docstring
+example, by referencing the key ``icepak``.
+
+.. code:: python
+
+    def assign_openings(self, air_faces):
+        """Assign openings to a list of faces.
+
+        Parameters
+        ----------
+        air_faces : list
+            List of face names.
+
+        Returns
+        -------
+        :class:`pyaedt.modules.Boundary.BoundaryObject`
+            Boundary object when successful or ``None`` when failed.
+
+        Examples
+        --------
+
+        Create an opening boundary for the faces of the "USB_GND" object.
+
+        >>> faces = icepak.modeler.primitives["USB_GND"].faces
+        >>> face_names = [face.id for face in faces]
+        >>> boundary = icepak.assign_openings(face_names)
+        pyaedt Info: Face List boundary_faces created
+        pyaedt Info: Opening Assigned
+
+        """
+
+Useful Features
+---------------
```

Comment:
Now that I think of it, we probably should create a directory with an `index.rst` and include different sections regarding ansys specific coding style and notes.  This document isn't too bad at the moment, but when kathy starts contributing, I think we'll have more to add.

We don't have to implement this now, but eventually we should have:

```
coding_style/
    index.rst
    coding_style.rst
    best_practices.rst
    docstring_examples.rst
```

where `docstring_examples.rst` contains..

```
Docstring Examples Best-Practices
=================================

General Notes
-------------------
Add general notes on how to write a docstring example.

Documentation Testing
---------------------

Setting Up ``doctest``
~~~~~~~~~~~~~~~~~~~~~~

Using ``pytest`` Fixtures
~~~~~~~~~~~~~~~~~~~~~~~~~

Using Ellipses For Random Output
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```